variables.mk

```diff
@@ -26,31 +26,45 @@ SBT_PROJECT ?= $(PROJECT)
 TB          ?= TestDriver
 TOP         ?= RocketTop
 
-# make it so that you only change 1 param to change most or all of them!
+#########################################################################################
+# subproject overrides
+# description:
+#   - make it so that you only change 1 param to change most or all of them!
+#   - mainly intended for quick developer setup for common flags
+#   - for each you only need to specify a CONFIG
+#########################################################################################
 SUB_PROJECT ?= example
+
+# for a BOOM based example system
 ifeq ($(SUB_PROJECT),boomexample)
-	# for a BOOM based system (provides all necessary params)
 	MODEL=BoomTestHarness
-	CONFIG=DefaultBoomConfig
 	TOP=BoomTop
 endif
+# for BOOM developers
 ifeq ($(SUB_PROJECT),boom)
-	# for BOOM developers (only need to provide a CONFIG)
 	PROJECT=boom.system
 	MODEL=TestHarness
 	CFG_PROJECT=boom.system
 	SBT_PROJECT=boom
 	TOP=ExampleBoomSystem
 endif
+# for Rocket-chip developers
 ifeq ($(SUB_PROJECT),rocketchip)
-	# for Rocket-chip developers
 	PROJECT=freechips.rocketchip.system
 	MODEL=TestHarness
-	CONFIG=DefaultConfig
 	CFG_PROJECT=freechips.rocketchip.system
 	SBT_PROJECT=rebarrocketchip
 	TOP=ExampleRocketSystem
 endif
+# for Hwacha developers
+ifeq ($(SUB_PROJECT),hwacha)
+	PROJECT=freechips.rocketchip.system
```

Comment:
Hwacha will want to use its own generator which adds its assembly tests and benchmarks.

Reply:
Ok. We will switch to using Hwachas generator